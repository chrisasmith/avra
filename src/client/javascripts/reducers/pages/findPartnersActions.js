import { api } from '../../services'
import { createRequests, doApiCall, createRequestTypes } from '../actionHelper'

export const GET_FIND_PARTNERS_PAGE = createRequestTypes('GET_FIND_PARTNERS_PAGE')
export const getFindPartnersPage = createRequests(GET_FIND_PARTNERS_PAGE)

export const FIND_PARTNERS_PAGE_DATA_SUCCESS = 'FIND_PARTNERS_PAGE_DATA_SUCCESS'
export function findPartnersPageDataSuccess(data) {
  return { type: FIND_PARTNERS_PAGE_DATA_SUCCESS, data }
}

export function getFindPartnersData() {
  return doApiCall({
    api: api.getFindPartnersData,
    action: getFindPartnersPage,
  })
}
