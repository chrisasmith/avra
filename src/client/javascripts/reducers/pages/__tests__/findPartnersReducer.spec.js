import reducer from '../findPartnersReducer'
import * as ActionTypes from '../findPartnersActions'

describe('Find Partners reducer', () => {
  it('should handle GET_FIND_PARTNERS_PAGE.SUCCESS', () => {
    expect(
      reducer({}, {
        type: ActionTypes.GET_FIND_PARTNERS_PAGE.SUCCESS,
        response: {
          data: [],
        },
      })
    ).toMatchSnapshot()
  })
})
