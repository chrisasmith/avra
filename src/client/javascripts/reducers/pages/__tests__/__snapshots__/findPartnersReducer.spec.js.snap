// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Find Partners reducer should handle GET_FIND_PARTNERS_PAGE.SUCCESS 1`] = `Object {}`;
